tests/contrib/kafka/test_kafka.py

```diff
@@ -0,0 +1,152 @@
+    return _producer
+
+
+@pytest.fixture
+def consumer(tracer):
+    _consumer = confluent_kafka.Consumer(
+        {
+            "bootstrap.servers": BOOTSTRAP_SERVERS,
+            "group.id": GROUP_ID,
+            "auto.offset.reset": "earliest",
+        }
+    )
+    Pin.override(_consumer, tracer=tracer)
+    _consumer.subscribe([TOPIC_NAME])
+    yield _consumer
+    _consumer.close()
+
+
+@pytest.mark.parametrize("tombstone", [False, True])
+@pytest.mark.snapshot
+def test_message(producer, consumer, tombstone):
+    if tombstone:
+        producer.produce(TOPIC_NAME, key=KEY)
+    else:
+        producer.produce(TOPIC_NAME, PAYLOAD, key=KEY)
+    producer.flush()
+    message = None
+    while message is None:
+        message = consumer.poll(1.0)
+
+
+@pytest.mark.snapshot(token="tests.contrib.kafka.test_kafka.test_service_override")
+def test_service_override_config(producer, consumer):
+    with override_config("kafka", dict(service="my-custom-service-name")):
+        producer.produce(TOPIC_NAME, PAYLOAD, key=KEY)
+        producer.flush()
+        message = None
+        while message is None:
+            message = consumer.poll(1.0)
+
+
+@pytest.mark.subprocess(env=dict(DD_KAFKA_SERVICE="my-custom-service-name"))
+@pytest.mark.snapshot(token="tests.contrib.kafka.test_kafka.test_service_override")
+def test_service_override_env_var():
+    import six
+
+    import ddtrace
+    from ddtrace.contrib.kafka.patch import patch
+    from ddtrace.contrib.kafka.patch import unpatch
+    from ddtrace.filters import TraceFilter
+
+    PAYLOAD = bytes("hueh hueh hueh", encoding="utf-8") if six.PY3 else bytes("hueh hueh hueh")
+
+    class KafkaConsumerPollFilter(TraceFilter):
+        def process_trace(self, trace):
+            # Filter out all poll spans that have no received message
+            return (
+                None
+                if trace[0].name == "kafka.consume" and trace[0].get_tag("kafka.received_message") == "False"
+                else trace
+            )
+
+    ddtrace.tracer.configure(settings={"FILTERS": [KafkaConsumerPollFilter()]})
+    patch()
+    import confluent_kafka
+
+    producer = confluent_kafka.Producer({"bootstrap.servers": "localhost:29092"})
+    consumer = confluent_kafka.Consumer(
+        {
+            "bootstrap.servers": "localhost:29092",
+            "group.id": "test_group",
+            "auto.offset.reset": "earliest",
+        }
+    )
+    ddtrace.Pin.override(producer, tracer=ddtrace.tracer)
+    ddtrace.Pin.override(consumer, tracer=ddtrace.tracer)
+    consumer.subscribe(["test_topic"])
+
+    producer.produce("test_topic", PAYLOAD, key="test_key")
+    producer.flush()
+    message = None
+    while message is None:
+        message = consumer.poll(1.0)
+    unpatch()
+    consumer.close()
+
+
+@pytest.mark.snapshot
+def test_analytics_with_rate(producer, consumer):
+    with override_config("kafka", dict(analytics_enabled=True, analytics_sample_rate=0.5)):
```

Comment:
This tests a sample rate with a single span being produced? Oh, or we're just checking that the tag is there essentially?

Reply:
This is a similar snapshot compared to the rest of the tests, with 1 `producer.produce()` and 1 `consumer.poll()` spans. But this one specifically is just making sure that the analytics sample rate is included and reflects the overridden config sample rate.